private/rules/maven_bom_fragment.bzl

```diff
@@ -12,9 +12,11 @@ MavenBomFragmentInfo = provider(
 )
 
 def _maven_bom_fragment_impl(ctx):
+    # Expand maven coordinates for any variables to be replaced.
+    coordinates = ctx.attr.maven_coordinates.format(**ctx.var)
```

Comment:
This line is actually fixing a bug introduced by https://github.com/bazelbuild/rules_jvm_external/pull/794 where the `<version>` in the generated pom isn't template expanded, might be better to move this to a separate PR since it's unrelated to this new feature

Reply:
sure will move to different PR